src/feedcontainer.cpp

```diff
@@ -289,4 +289,13 @@ unsigned int FeedContainer::unread_item_count() const
 	return unread_guids.size();
 }
 
+void FeedContainer::replace_feed(unsigned int pos,
+	std::shared_ptr<RssFeed> feed)
+{
+	assert(pos < feeds.size());
```

Comment:
It looks somewhat strange to have the assert before locking `feeds_mutex`.
I expect it is possible for the controller to call `FeedContainer::set_feeds(...)` in a way which lowers the amount of feeds (in-between the assert and the locking of the `feeds_mutex`).

Reply:
Fixed by https://github.com/newsboat/newsboat/pull/1166/commits/d570f32610780e975d61bccbe0dc60bd5d8f5341.